File: project/server/Models/Comments/CommentDto.cs
using server.Models.Categories;
using server.Models.Locations.DTOs;
using server.Models.Projects.DTOs;
using server.Types.Enums;

namespace server.Models.Comments
{
	public record class CommentDto
	{
		public Guid Id { get; set; }
		public ProjectDto? Project { get; set; }
		public LocationDto? Location { get; set; }
		public Guid? CategoryId { get; set; }
		public CategoryDto? Category { get; set; }
		public CommentType Type { get; set; }
		public string Content { get; set; } = string.Empty;
	}
}

File: client/src/types/dtos/comment-dto.ts
import { CommentType } from "./comment-type.ts";
import type CategoryDto from "./category-dto.ts";
import type ProjectDto from "../shared/project-dto.ts";
import type LocationDto from "./location-dto.ts";

export default interface CommentDto {
	id: string | null;
	project?: ProjectDto;
	location: LocationDto;
	categoryId: string | null;
	category?: CategoryDto;
	type: CommentType;
	content: string;
	// Threading fields
	rootCommentId: string | null;
	parentCommentId: string | null;
	depth?: number;
	replies?: CommentDto[];
}
